pinot-tools/src/main/java/com/linkedin/pinot/tools/AutoAddInvertedIndex.java

```diff
@@ -0,0 +1,212 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.tools;
+
+import com.linkedin.pinot.common.config.AbstractTableConfig;
+import com.linkedin.pinot.common.config.IndexingConfig;
+import com.linkedin.pinot.common.data.Schema;
+import com.linkedin.pinot.common.utils.SchemaUtils;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.apache.helix.PropertyPathConfig;
+import org.apache.helix.PropertyType;
+import org.apache.helix.ZNRecord;
+import org.apache.helix.manager.zk.ZKHelixAdmin;
+import org.apache.helix.manager.zk.ZNRecordSerializer;
+import org.apache.helix.store.zk.ZkHelixPropertyStore;
+import org.json.JSONObject;
+
+
+/**
+ * This tool will directly modify the table configs in ZooKeeper.
+ * PLEASE USE IT VERY CAREFULLY!
+ */
+public class AutoAddInvertedIndex {
+  private AutoAddInvertedIndex() {
+  }
+
+  private static final String AUTO_GENERATED_INVERTED_INDEX_KEY = "autoGeneratedInvertedIndex";
+
+  private static final String ZK_ADDRESS = "zkAddress:zkPort";
+  private static final String CLUSTER_NAME = "pinot-cluster";
+  private static final String TABLE_CONFIG_PATH = "/CONFIGS/TABLE/";
+  private static final String SCHEMA_PATH = "/SCHEMAS/";
+  private static final ZKHelixAdmin HELIX_ADMIN = new ZKHelixAdmin(ZK_ADDRESS);
```

Comment:
ZK_ADDRESS is hardcoded for this initial version. I will make it an argument passed in by the user.